tests/common.go

```diff
@@ -1869,7 +1892,147 @@ func ChangeNamespaces(contexts []*scheduler.Context,
 	return nil
 }
 
-func updateNamespace(in interface{}, namespaceMapping map[string]string) error {
+func CloneSpec(spec interface{}) (interface{}, error) {
```

Comment:
How about giving the implementation of CloneSpec below a try? I think it could make the code more readable, but please keep in the mind that I have not tested it.

```
func CloneSpec(spec interface{}) (interface{}, error) {
	var supportedTypes = map[reflect.Type]interface{}{
		reflect.TypeOf(&appsapi.Deployment{}): &appsapi.Deployment{},
		reflect.TypeOf(&appsapi.StatefulSet{}): &appsapi.StatefulSet{},
		.
		.
		.
		reflect.TypeOf(&admissionregistrationv1.ValidatingWebhookConfiguration{}): &admissionregistrationv1.ValidatingWebhookConfiguration{},
	}
    specType := reflect.TypeOf(spec)
    if clone, ok := supportedTypes[specType]; ok {
        if webhookConfig, ok := clone.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
			var webhooks []admissionregistrationv1.ValidatingWebhook
			for i, webhook := range webhookConfig.Webhooks {
			    serviceClone := *spec.(*admissionregistrationv1.ValidatingWebhookConfiguration).Webhooks[i].ClientConfig.Service
			    webhook.ClientConfig.Service = &serviceClone
			    webhooks = append(webhooks, webhook)
			}
	        clone.(*admissionregistrationv1.ValidatingWebhookConfiguration).Webhooks = webhooks
        } else {
            reflect.ValueOf(clone).Elem().Set(reflect.ValueOf(spec).Elem())
        }
        return clone, nil
    }
    return nil, fmt.Errorf("unsupported object while cloning: %v", specType)
}

```

Reply:
How about adding a docstring that describes what the function does?

Reply:
docstring done.
reflect based solution doesn't seem possible as it is not working. Thanks for exploring the possibility, though.

Reply:
✅